### ISSUE TITLE
Missing static libraries on sdk created by bitbake -c populate_sdk

### ISSUE THREAD

**Issue by alitofresko** [aws/aws-iot-device-sdk-cpp-v2#306]
Confirm by changing [ ] to [x] below:
- [ x] I've searched for [previous similar issues](https://github.com/aws/aws-iot-device-sdk-cpp-v2/issues/) and didn't find any solution

**Known Issue**
- [ x] I'm using ATS data type endpoint: the endpoint should look like `<prefix>-ats.iot.<region>.amazonaws.com`

**Platform/OS/Device**
Raspberry pi cm4

**Describe the question**
I'm trying to setup a development environment for a custom aws-iot-device-client.
I create my sysroot with bitbake image-name -c populate_sdk.
The resulting sdk missing for static libraries and the project I'm trying to build (cmake + qtcreator) fails due to missing static librarias (libcrypto the first error)

What is the correct way to setup an environment able to debug a custom aws-iot-device-client on a remote raspberry cm4 device  with a custom image?

**Comment by jmklix** [comment]
What error/s are you getting? and can you include detailed steps to reproduce this.

I haven't ever tried using a raspberry pi cm4, but you might also find it useful to look at the [meta-aws](https://github.com/aws/meta-aws)

As for setting up debug environment, you can set up VS Code to ssh into it and then debug from there.

**Comment by alitofresko** [comment]
Steps to reproduce the issue are so many. I'll try to explain a summary:
git clone -b dunfell git://git.yoctoproject.org/poky.git poky-dunfell
cd poky-dunfell
git clone -b dunfell git://git.openembedded.org/meta-openembedded
git clone -b dunfell git://git.yoctoproject.org/meta-raspberrypi
git clone -b dunfell git://git.yoctoproject.org/meta-security.git
git clone -b dunfell https://github.com/aws/meta-aws.git
Create a recipe image with aws-iot-device-sdk-cpp-v2
Create an sdk for development with:
bitbake image -c populate_sdk

The error comes building https://github.com/awslabs/aws-iot-device-client using the resulting sdk.
The error coming during cmake

source environment-setup-sdk
mkdir build && cd build
cmake ../aws-iot-device-client/
The error was:
Could NOT find LibCrypto (missing: LibCrypto_LIBRARY)

However I suspect that the workflow to develop on a target linux device with aws-iot-device-sdk-cpp-v2 is different from my approach.
Please help me to find out the right way.

**Comment by alitofresko** [comment]
It seems that aws-iot-device-sdk-cpp-v2 recipe do not create right development packages needed by populate_sdk task.
Should I post the same question to the meta-aws repository?

**Comment by bretambrose** [comment]
I don't know anything about bitbake or what it does, but in general, even with a successfully built sdk (and I don't know what "create a recipe image with aws-iot-device-sdk-cpp-v2" actually does), you always need to point cmake at the install location via CMAKE_PREFIX_PATH unless you literally install to the system directories (not at all advised, especially given that we build our own libcrypto and you'd be stomping system openssl install).

**Comment by alitofresko** [comment]
Thankyou very much for your suggestion. 
However CMAKE_PREFIX_PATH is set-up correctly.
I think theat the issue depends on how aws-iot-device-sdk-cpp-v2.bb recipe has been written.
I solved following the example aws-iot-device-client. 
During development, the cmake project, as a default, build aws-iot-device-sdk-cpp-v2 sdk  as a subproject (BUILD_SDK=ON) so there's no need to have libraries and include file into my custom sdk (created by populate_sdk task)
The problem related to missing libcrypto library has been solved appending staticdev-pkgs to SDKIMAGE_FEATURES to my recipe image.
Thanks again for your support.
